openquake/calculators/base.py

```diff
@@ -436,7 +436,7 @@ def pre_execute(self):
             # there is a precalculator
             precalc_id = self.oqparam.hazard_calculation_id
             self.precalc = (self.compute_previous() if precalc_id is None
-                            else self.read_previous(precalc_id))
+                            else self.read_previous(int(precalc_id)))
```

Comment:
The cast should not be needed here. The bug has to be in another places. I have to look at it.